hail/python/hailtop/aiocloud/aiogoogle/client/storage_client.py

```diff
@@ -630,22 +635,28 @@ def schemes() -> Set[str]:
     def storage_location(self, uri: str) -> str:
         return self.get_bucket_and_name(uri)[0]
 
-    async def is_hot_storage(self, location: str) -> bool:
+    async def is_hot_storage(self, location: str, uri: str) -> bool:
```

Comment:
I think it's worth moving the added functionality here into a distinct method. Adding to this method is starting to confuse this signature for me. Is it on the caller to ensure that `uri` is derived from `location`? Can the bucket be hot storage by default but the object was explicitly moved to cold storage / what happens then?